sys/freebsd/sys.txt

```diff
@@ -197,6 +210,20 @@ rtprio {
 	prio	int16[0:RTP_PRIO_MAX]
 }
 
+thr_param {
+	start_func	ptr[in, text[target]]
+	arg		intptr
+	stack_base	ptr64[in, array[int8]]
+	stack_size	bytesize[stack_base, intptr]
+	tls_base	ptr64[in, array[int8]]
+	tls_size	bytesize[tls_base, intptr]
+	child_tid	ptr[out, lwpid]
+	lwpid		ptr[out, lwpid]
```

Comment:
These are actually pointers to long, but lwpid is (correctly) defined as int32.

Reply:
I had noticed this when reviewing pre-submission but apparently didn't go back to fix it -- now that I look at it more, I'm kind of wondering: my limited understanding of this was that syzkaller would take this definition and spawn threads then use the resulting lwpid in subsequent thr_* calls (lwpid being an output resource). Is that still the case if the type's "wrong"?

I guess I could always solve that by adding in a fake struct that has just an lwpid lwpid and [align[PTR_SIZE]], if it's a problem.

Reply:
A fake struct would work. Also a struct that is always larger than long (e.g. lwpid + fake intptr field) should work too.

But I think it should work fine as is. It will only fail if the lwpid is at the very end of a page and the next page is unmapped, right? But that's unlikely and is an interesting scenario to test :)
Or does kernel check that the padding is 0? If so, then it can make sense to add a fake field afterwards.

Reply:
AFAICT the kernel will just suword() into it, so the main concerns are if we can't cope with the unaligned store and maybe if we were to start fuzzing, e.g., ppc64, which is BE by default on FreeBSD.

Reply:
Yes, we might get errors if the output fields are not sufficiently aligned, but for supported architectures the kernel will cope with it. I'm more worried that the extra four byte write will clobber something else and break the input program somehow.